### PR TITLE
[WIP] Migrate website_logo to v9

### PR DESCRIPTION
**Pull request by lasley**
Tested & implemented suggestion from #190 + code upgraded to new OCA guidelines.

### CODE REVIEW

File: website_logo/README.rst [hunk 1]
```diff
@@ -1,20 +1,22 @@
 .. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
     :alt: License: AGPL-3
 
-Website logo
+Website Logo
```
Comment (pedrobaeza) [inline website_logo/README.rst:6]: Follow lates README template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: website_logo/views/company_view.xml [hunk 2]
```diff
@@ -6,11 +6,11 @@
             <field name="model">res.company</field>
             <field name="inherit_id" ref="base.view_company_form"></field>
             <field name="arch" type="xml">
-                <page string="Configuration" position="inside">
```
Comment (pedrobaeza) [inline website_logo/views/company_view.xml:9]: You can use now in v9 the `<odoo>` tag instead `<openerp>` and remove the `<data>` tag

Comment (lasley) [inline website_logo/views/company_view.xml:9]: Whaaaaaat?! That's amazing! Does this carry over to all xml files, or just views?

**Comment by lasley** [comment]
Ouch @coveralls - I'll add tests 😉

File: website_logo/views/company_view.xml (hunk 2 above)
Comment (pedrobaeza) [inline website_logo/views/company_view.xml:9]: All XML files that uses noupdate="0"

Comment (lasley) [inline website_logo/views/company_view.xml:9]: ❤️  thanks for the insight!

**Comment by lasley** [comment]
I have no idea how to get these tests to pass. I have tried everything from a standard `.write` on the company, to straight up SQLing the logo in the right place. 

It looks like OpenERP decided to not test this section of their code either; maybe the same problems? 

Regardless, I have rebased away from my tests if we want to merge as is. I'm open to pointers as well, if a better Odoo-er than me wants to take a look - https://github.com/laslabs/website/blob/feature/9.0/website_logo_test/website_logo/tests/test_logo.py

**Comment by pedrobaeza** [comment]
Controllers are usually tested with PhantomJS tests, but you can import and call directly the method as an alternative method, but you will have to deal with the answer as an object.

**Comment by lasley** [comment]
Interesting, I thought I was using Phantom helpers in the HttpCase with the `url_open`. TIL, thanks @pedrobaeza I will try your suggestions 😄

**Comment by lasley** [comment]
Alright woot got it! 

I wrote them in Phantom at first, but wasn't getting the coverage I was looking for; so I reverted to the good ol' mock pattern. It seems the Phantom tests are designed pretty heavily around the tours, which somewhat conflicted with the scope of this module. 

This is ready IMO

File: website_logo/tests/test_logo.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+# © 2015-TODAY LasLabs Inc.
```
Comment (pedrobaeza) [inline website_logo/tests/test_logo.py]: Please put only 2016, because TODAY is relative :wink:

**Comment by pedrobaeza** [comment]
:+1:

File: website_logo/README.rst [hunk 4]
```diff
@@ -1,29 +1,52 @@
...
+======================
+
 
-Note that, in case of anonymous visitors, the company associated to admin user will be used
 
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues
+<https://github.com/OCA/{project_repo}/issues>`_. In case of trouble, please
```
Comment (eLBati) [inline website_logo/README.rst]: Please replace `project_repo`

**Comment by eLBati** [comment]
@lasley looking at the diff from 8.0 version
http://pastie.org/10809125
it seems some changes are missing, like https://github.com/OCA/website/commit/1616b126dda826632f4c33668b90acd4ec8e7d8c

Please add them, thanks

**Comment by lasley** [comment]
Wow I really need to do better at following up on my PRs. 

@eLBati & @pedrobaeza - Sorry if this should be more clear, but is there a way for me to extract pieces of a commit in order to be included in this PR? The only way I can think of to bring commits such as c3bd993138041c0b438be6d20f3be98fdd2f669c in is by literally copying the diff and creating a new commit. I feel like there should be a more elegant way here, like some sort of partial cherry pick, but I am totally at a loss.

**Comment by pedrobaeza** [comment]
This is what I do on 9.0 branch (you can use the migration guide I made for 9.0 as inspiration: https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-9.0):

```
git fetch origin
git checkout origin/8.0
git checkout -b 8.0-extract
git filter-branch --subdirectory-filter website_logo
git filter-branch -f --tree-filter 'mkdir -v website_logo ; git mv -k * website_logo' HEAD
```

Now I have the module isolated in one branch. I see the log (`git log`) and copy the commit numbers of the OCA Transbot. I go now again to the 9.0 branch, and make `git cherry-pick <commit>` for each of the commits I want to bring, but now I'm sure this commit only makes modifications over my module.

**Comment by lasley** [comment]
This is amazing, thank you @pedrobaeza - this is really going to help me move all my vertical medical PRs up too (we're about 200 commits ahead of the PRs at this point, and I was honestly dreading that rebase nightmare).

I will play around, and update this PR with the results. Thanks again, as always 😄

**Comment by lasley** [comment]
Well that was some wizardry. Sorry about the comment spam at first, I didn't realize that I wasn't on my test branch when pushing 😑 

Should be gtg here!

File: website_logo/migrations/8.0.2.0.0/post-migration.py [hunk 5]
```diff
@@ -0,0 +1,34 @@
+# -*- coding: utf-8 -*-
```
Comment (lmignon) [inline website_logo/migrations/8.0.2.0.0/post-migration.py]: migration scripts are not relevant here since they are for 8.0

**Comment by lasley** [comment]
Migrations removed

**Comment by rafaelbn** [comment]
Rebuilding for testing

File: website_logo/README.rst [hunk 6]
```diff
@@ -1,29 +1,64 @@
...
+internal company logo (for reports) and a different website logo.
 
 
 Configuration
 =============
 
 To configure this module, you need to:
 
-* go to your company and load a logo image in the website logo field
+* Go to 'Settings' > 'Configuration' > 'Website Settings'
```
Comment (rafaelbn) [inline website_logo/README.rst]: This instructions are not valid in v9. Please change where you can change logo.

**Comment by rafaelbn** [comment]
Hi @lasley , I don't see any please to change logo:

**Comment by lasley** [comment]
Hi @rafaelbn - Thanks for the review. ReadMe instructions updated to correspond, although I am thinking these might have been invalid for v8 too.

For easy reference:
- Navigate to Company Settings, such as by clicking logo on left nav bar
- Click the Configuration tab
- Load a logo image in the field under the Website logo header

**Comment by lasley** [comment]
Hmmm so after comparing deeper between current v8 and this branch, I think maybe the rebase swapped some things. I am going to tag this with a WIP, and will be able to take a look sometime this week.

**Comment by pedrobaeza** [comment]
Is this completed?

**Comment by lasley** [comment]
TBH I can't remember. It seems to work in my local, ~~but I haven't been able to get Runbot to build. Do we have some sort of active issue with Runbot at the moment?~~ Hmmm ok maybe the status was just incredibly slow to update. I'll double check this and let everyone know.

**Comment by pedrobaeza** [comment]
It seems that it's working properly now, so no clue.

**Comment by eLBati** [comment]
Hello @lasley 
please also check travis:

```
Error details:
View inheritance may not use attribute 'string' as a selector.
```

**Comment by lasley** [comment]
Superseded by #245

**Pull request closed without merging**